pkg/cmd/render/render.go

```diff
@@ -227,15 +237,47 @@ func (r *renderOpts) Run() error {
 	return WriteFiles(&r.generic, &templateData.FileConfig, templateData)
 }
 
+// setBootstrapIP gets a list of IPs from local interfaces and returns the first IPv4 or IPv6 address.
+func (t *TemplateData) setBootstrapIP() error {
```

Comment:
style-wise, we produce getters for this information and then assign the values.  We don't make mutating functions in general.